boss-ws/src/main/java/org/genomebridge/boss/http/service/DatabaseBossAPI.java

```diff
@@ -398,6 +398,26 @@ private String getMessage(String key) {
         return msg;
     }
 
+    @Override
+    public ErrorDesc getResumableUploadURL(String objectId, String userName, ResolveResponse resp) {
+        ErrorDesc error = null;
+        if ( userName == null  ){
+            return badReqErr(getMessage("remoteUser"));
+        }
+        BossDAO dao = getDao();
+        ObjectRow rec = dao.findObjectById(objectId);
+        if ( rec == null ) {
+            return  notFoundErr(objectId);
+        }
+        if ( !dao.canWrite(objectId,userName) ) {
+            return  writePermsErr(objectId,userName);
+        }
+        ObjectStore objStore = getObjectStore(StoragePlatform.CLOUDSTORE.getValue());
```

Comment:
We need to use the object store indicated by the ObjectRow -- we can't hard-code CLOUDSTORE here.  But one more thing:  we need to check to make sure that the rec.storagePlatform is not OPAQUEURI because the getObjectStore method returns null for that type.  The needed code is on lines 268-9 in this file.
And let's move this method up with the other @Override public methods.